packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts

```diff
@@ -138,53 +140,63 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
     }
   }
 }
-export class LibraryRetrieveSourcePathExecutor extends LibraryCommandletExecutor<
+export class LibraryRetrieveSourcePathExecutor extends RetrieveExecutor<
   LocalComponent[]
 > {
   private openAfterRetrieve: boolean;
 
   constructor(openAfterRetrieve = false) {
-    super('Retrieve (Beta)', 'force_source_retrieve_beta', OUTPUT_CHANNEL);
+    super(
+      nls.localize('force_source_retrieve_text'),
+      'force_source_retrieve_beta'
+    );
     this.openAfterRetrieve = openAfterRetrieve;
   }
 
-  public async run(
+  protected async getComponents(
     response: ContinueResponse<LocalComponent[]>
-  ): Promise<boolean> {
-    const dirPath = (await SfdxPackageDirectories.getDefaultPackageDir()) || '';
-    const output = path.join(getRootWorkspacePath(), dirPath);
-    const comps: LocalComponent[] = response.data;
-
-    const components = new ComponentSet(
-      comps.map(lc => ({ fullName: lc.fileName, type: lc.type }))
+  ): Promise<ComponentSet> {
+    const filter = new ComponentSet(
```

Comment:
I'm not able to walkthrough this portion of the code - I can't find a way to fire it. Is that because this class `LibraryRetrieveSourcePathExecutor` is also defined in `forceSourceRetrieveSourcePath`? Let me know if this is just simple user error on my end.

Reply:
It looked like the index was only referring to the one defined within `forceSourceRetrieveSourcePath`.

Reply:
This command is only used by the org browser (right now), so there isn't a command palette option for it. Try retrieving components through that tree.